databases_shiny_apps/connect-access-db.R

```diff
@@ -0,0 +1,74 @@
+library(tidyverse) #This package helps import spreadsheet-formatted data, primarily from Excel
+# library(data.table) #Helps with data management
+# library(dbplyr) #This package will help interface with databases by translating R code into database-specific variants
+# library(RODBC)
+library(odbc)
+library(DBI)
+# library(sqldf) # This package might help with more sophisticated SQL queries
+
+
+### with RODBC
+# connection_access <- odbcConnectAccess() ## only available for 32-bit version of Windows.
+# connection <- odbcConnect("<DSN>")
+# query <- "<SQL Query>"
+# data <- sqlQuery(connection, query, )
+# str(data)
+# odbcCloseAll() # At the end of an R session please remember to close the connection. 
+
+#ODBC -  Open Database Connectivity  a standard application programming interface (API) for accessing database management systems
+#DSN - File data sources store the connection information in a text file  not the Windos registry 
+
+
+# Finding your ODBC drivers https://support.office.com/en-us/article/administer-odbc-data-sources-b19f856b-5b9b-48c9-8b93-07484bfab5a7
+
+
+### with odbc
+odbcListDrivers() # to get a list of the drivers your computer knows about 
+con <- dbConnect(odbc::odbc(), "2019_CREP_Database")
+dbListTables(con) # To get the list of tables in the database
+
+odbc_result <- DBI::dbReadTable(con, "Fish_Abundance") #The dbReadTable() command will return the data table from the database as class: "data.frame"
+result_dbi<- as_tibble(odbc_result) # Use as_tibble() turn the dataframe as a tibble dataframe to continue with queries using the dplyr syntax
+result_dplyr <- as_tibble(tbl(con, "Fish_Abundance")) # OR directly return the data table from the database as a tibble
```

Comment:
```suggestion
odbc_result <- DBI::dbReadTable(con, "Fish_Abundance")
result_dbi<- as_tibble(odbc_result)
result_dplyr <- as_tibble(tbl(con, "Fish_Abundance"))
```

Having comments that just describe what the code is doing rather than explain why the code is doing something is generally considered bad practice as it means the code isn't clear enough as is.